### PR TITLE
modules/shared/builder: add gc.gbFree option

### PR DESCRIPTION
**Pull request by zowoq**

### CODE REVIEW

File: modules/shared/builder.nix [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
 {
-  nix.gc.options = ''
-    --max-freed "$((50 * 1024**3 - 1024 * $(df -P -k /nix/store | tail -n 1 | ${pkgs.gawk}/bin/awk '{ print $4 }')))"
+  options.nixCommunity.gc.min = lib.mkOption {
+    type = lib.types.int;
+    default = 50;
+  };
+
+  config.nix.gc.options = ''
+    --max-freed "$((${toString config.nixCommunity.gc.min} * 1024**3 - 1024 * $(df -P -k /nix/store | tail -n 1 | ${pkgs.gawk}/bin/awk '{ print $4 }')))"
```
Comment (Mic92) [inline modules/shared/builder.nix:9]: Maybe some comment what this line does?

File: modules/shared/builder.nix [hunk 2]
```diff
@@ -1,6 +1,11 @@
-{ pkgs, ... }:
+{ config, lib, pkgs, ... }:
 {
-  nix.gc.options = ''
-    --max-freed "$((50 * 1024**3 - 1024 * $(df -P -k /nix/store | tail -n 1 | ${pkgs.gawk}/bin/awk '{ print $4 }')))"
+  options.nixCommunity.gc.min = lib.mkOption {
+    type = lib.types.int;
+    default = 50;
```
Comment (Mic92) [inline modules/shared/builder.nix:5]: Is this percent?

Comment (zowoq) [reply modules/shared/builder.nix:5]: GB.

File: modules/shared/builder.nix (hunk 1 above)
Comment (zowoq) [reply modules/shared/builder.nix]: I'll leave this for now.

Comment (Mic92) [reply modules/shared/builder.nix]: the module description makes it better.

**Pull request merged**